lib/std/Thread.zig

```diff
@@ -849,7 +849,7 @@ const WasiThreadImpl = struct {
             args: @TypeOf(args),
             fn entry(ptr: usize) void {
                 const w: *@This() = @ptrFromInt(ptr);
-                @call(.auto, f, w.args);
+                @call(.auto, f, w.args) catch unreachable;
```

Comment:
This removes the ability to pass a plain `fn () void` function to `spawn`. I suppose we could just re-use [callFn](https://github.com/ziglang/zig/blob/9b1ea46324a281b5914b6541d989e500fb0cda6b/lib/std/Thread.zig#L397) here. Would you mind adding that? Unfortunately, I cannot remember why I didn't choose to use that here in the initial implementation.

Reply:
Sure!

But there's another regression. 

`wasmtime` fails with `wasi::thread-spawn` has not been defined`, but that may be due to a change in `wasmtime`. 

And `wasmer` and `w2c2` fail with an `Out of memory` error, which is super weird.

Reply:
Sounds like it's missing the flags `--wasm-features threads --wasi-modules experimental-wasi-threads` for `wasmtime`. When I try this on the latest `wasmtime` I seem to be getting the same `Out of memory` error as you were experiencing on the other runtimes, so I suppose this code did indeed regress. Unfortunately, we do not yet test this feature as part of Zig's test suite as the feature itself is experimental and requires a ton of flags we currently do not pass to either Zig or Wasmtime. If this is something you're interested in resolving let me know. If not, we can get this back into a compiling state and we can look into resolving the regression in a separate PR.